libs/bart_derive/src/parsbart.rs

```diff
@@ -25,18 +25,36 @@ struct CloseScopeTag {
     pub name: Name,
 }
 
+fn identifier(input: &str) -> IResult<&str, &str> {
+  let input_length = input.input_len();
+  if input_length == 0 {
+    return IResult::Incomplete(Needed::Unknown);
+  }
+
+  for (idx, item) in input.iter_indices() {
+    if !(item.is_alphanum() || item == '_') {
+      if idx == 0 {
+        return IResult::Error(error_position!(ErrorKind::AlphaNumeric, input))
+      } else {
+        return IResult::Done(input.slice(idx..), input.slice(0..idx))
+      }
+    }
+  }
+  IResult::Done(input.slice(input_length..), input)
+}
```

Comment:
This function is written with two-space indents. Mind converting to four-space to match the surrounding code?